### PR TITLE
Utilities/fortune: Add `-w` flag to wait based on message length

### PR DESCRIPTION
**Pull request by ninadsachania**
The `-w` flag in the fortune utility introduces a delay before termination. The wait time is proportional to the number of characters in the fortune message.

The `-w` flag is in the BSD version of `fortune`. Here's the [proof](https://man.freebsd.org/cgi/man.cgi?query=fortune).

### CODE REVIEW

File: Userland/Utilities/fortune.cpp [hunk 1]
```diff
@@ -78,6 +90,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     Optional<bool> force_color;
 
+    bool wait_before_termination = false;
+    int const minimum_secs_to_wait = 6;
+    int const chars_per_second = 20;
```
Comment (timschumi) [inline Userland/Utilities/fortune.cpp:95]: These could even be `constexpr` (and unsigned).

File: Userland/Utilities/fortune.cpp [hunk 2]
```diff
@@ -151,5 +168,8 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     if (stdout_is_tty)
         outln(); // Tasteful spacing
 
+    if (wait_before_termination)
+        sleep((unsigned int)max(chosen_quote.length() / chars_per_second, minimum_secs_to_wait));
```
Comment (timschumi) [inline Userland/Utilities/fortune.cpp:172]: The cast here should not be C-style. In fact, it is not even required anymore once the type of the constants is corrected.

File: Userland/Utilities/fortune.cpp [hunk 3]
```diff
@@ -46,6 +46,18 @@ class Quote {
...
     Optional<ByteString> const& context() const { return m_context; }
 
+    size_t length() const
+    {
+        // TODO: Should we consider m_utc_time's length as well?
+        size_t total = 0;
+
+        total = m_quote.length() + m_author.length() + m_url.length();
+        if (m_context.has_value())
+            total += m_context.value().length();
```
Comment (timschumi) [inline Userland/Utilities/fortune.cpp:56]: I _think_ `m_context->length()` should work, but I can't confirm it right now. It's also a very minor nitpick.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Userland/Utilities/fortune.cpp (hunk 3 above)
Comment (ninadsachania) [reply Userland/Utilities/fortune.cpp:56]: I tried that and it didn't work. Crashed when `m_context` didn't have a value.

File: Userland/Utilities/fortune.cpp (hunk 1 above)
Comment (ninadsachania) [reply Userland/Utilities/fortune.cpp]: Done.

**Comment by nico** [comment]
Independent of implementation comments: Why do you want this?

The usual recommendation is for people to try to use the system and fix things they run into. I have a hard time imagining that you need this flag. If that's wrong, then apologies and we can merge some version of this (maybe you can describe your use case then?) Otherwise, as mentioned in other PRs, maybe try doing more interesting things 🙂 (the Screenshot change is great for example! If you can't think of other things to do, someone on discord started making Screenshot to animated screen recordings a while back but didn't finish that; maybe that's something you could look into?)

File: Userland/Utilities/fortune.cpp (hunk 3 above)
Comment (timschumi) [reply Userland/Utilities/fortune.cpp:56]: The `.has_value()` check still needs to be there, yes, but the very verbose `.value().` can be replaced with `->`.

**Pull request closed without merging**